### PR TITLE
search for loadlocationtype when calling defineclass

### PR DESCRIPTION
**Pull request by tajila**
search for loadlocationtype when calling defineclass

When calling Class.forname() it uses ClassLoader.defineclass() to load
the class. In this case we never search for where the class was loaded
from and fill the J9TranslationLocalBuffer. This means that when we are
creating the ramClass the loadLocation is set to LOAD_LOCATION_UNKNOWN
and we will not bother to search for the module that the class belongs
to, instead we will assign it to the unamedModule.

This will cause problems as classes which are loaded from modules will
have the incorrect module in the J9Class.

This change ensures that we always search for the class location, if the
class cannot be found we leave the location as LOAD_LOCATION_UNKNOWN.

Also removed the obsolete J2ME flags

Closes: #110

Signed-off-by: tajila <atobia@ca.ibm.com>

Reviewers: @JasonFengJ9 @gacholio

### CODE REVIEW

**Comment by tajila** [comment]
@JasonFengJ9 please review these changes

File: runtime/jcl/common/jcldefine.c [hunk 1]
```diff
@@ -145,6 +145,10 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 	tempClassBytes = classBytes;
 	tempLength = length;
 
+	/* Try to find classLocation. Ignore return code because there are valid cases where it might not find it (ie. bytecode spinning).
+	 * If the class is not found the default class location is fine. */
```
Comment (gacholio) [inline runtime/jcl/common/jcldefine.c]: */ should be on it's own line, with stars lined up

**Comment by gacholio** [comment]
Fails win64 compiles.

http://vmfarm.ottawa.ibm.com/jobs_by_status.php?build_id=367157&status=FAILED

../common/jcldefine.c(151) : warning C4244: 'function' : conversion from 'UDATA' to 'U_32', possible loss of data

Other compile failures appear unrelated.

**Comment by tajila** [comment]
Im working on fixing the compilation failure in https://github.com/eclipse/openj9/pull/327

**Comment by JasonFengJ9** [comment]
Sorry for late response. 
Just wondering if there is a simplified testcase to show the problem?

**Pull request merged**